packages/react-components/src/box/src/Box.tsx

```diff
@@ -0,0 +1,48 @@
+import { ElementType, ForwardedRef, ReactElement, ReactNode, forwardRef } from "react";
+import { InnerPropsToProps, omitProps } from "../../shared";
+
+export interface InnerBoxProps {
+    /**
+    * An HTML element type or a custom React element type to render as.
+    */
+    as?: ElementType;
+    /**
+     * [Slot](?path=/docs/getting-started-slots--page) to render into.
+     */
+    slot?: string;
+    /**
+     * React children
+    */
+    children?: ReactNode;
+    /**
+    * @ignore
+    */
+    forwardedRef: ForwardedRef<any>
+}
+
+export type BoxProps = InnerPropsToProps<InnerBoxProps>;
```

Comment:
InnerBoxProps is used by the inner component and the storybook documentation generation. It contains a forwardedRef, but no ref attributes.

BoxProps is the resulting props interface for the box component. It contains a ref attribute, and no forwardedRef

Reply:
As I wrote in another PR I would prefer if we could somehow do this without any custom code, only native TS. It's too abstract for me.